parasolr/query.py

```diff
@@ -158,6 +187,34 @@ def filter(self, *args, **kwargs) -> 'SolrQuerySet':
 
         return qs_copy
 
+    def facet(self, *args: str, **kwargs) -> 'SolrQuerySet':
+        """
+        Request facets for specified fields. Returns a new SolrQuerySet
+        with Solr faceting enabled and facet.field parameter set. Does not
+        support ranged faceting.
+
+        Subsequent calls will reset the facet.field to the last set of
+        args in the chain.
+
+        For example::
+
+            qs = queryset.facet('person_type', 'age')
+            qs = qs.facet('item_type')
+
+        would result in `item_type` being the only facet field.
+        """
+        qs_copy = self._clone()
+
+        # enable faceting
+        qs_copy.facet_opts['facet'] = True
```

Comment:
Can we just infer `facet=true` if `facet_opts` is not empty when we generate query opts?